tests/test_mutations.py

```diff
@@ -472,57 +821,464 @@ def test_keep_mutation_parent_zero_rate(self):
+                alleles = [a.decode() for a in model.alleles]
+                num_alleles = len(alleles)
+                roots = np.zeros((num_alleles,))
+                transitions = np.zeros((num_alleles, num_alleles))
+                for site in ts.sites():
+                    aa = site.ancestral_state
+                    roots[alleles.index(aa)] += 1
+                for mut in ts.mutations():
+                    if mut.parent == tskit.NULL:
+                        pa = ts.site(mut.site).ancestral_state
+                    else:
+                        pa = ts.mutation(mut.parent).derived_state
+                    da = mut.derived_state
+                    transitions[alleles.index(pa), alleles.index(da)] += 1
+                num_muts = np.sum(roots)
+                if verify_roots:
+                    exp_roots = num_muts * model.root_distribution
+                    self.chisquare(roots, exp_roots)
+                for j, (row, p) in enumerate(zip(transitions, model.transition_matrix)):
+                    p[j] = 0
+                    p /= sum(p)
+                    self.chisquare(row, sum(row) * p)
+
+    def verify_mutation_rates(self, model):
+        # this test only works if the probability of dropping a mutation
+        # doesn't depend on the previous state
+        assert(len(set(np.diag(model.transition_matrix))) == 1)
+        np.random.seed(23)
+        ots = msprime.simulate(10, random_seed=5, recombination_rate=0.05, length=20)
+        for discrete in (False, True):
+            for rate in (1, 10):
+                print("::::", discrete, rate)
+                mut_rate = rate * (1 - model.transition_matrix[0, 0])
+                ts = msprime.mutate(ots, random_seed=6, rate=rate, model=model,
+                                    discrete=discrete)
+                lower_pvals = np.zeros(ts.num_trees)
+                upper_pvals = np.zeros(ts.num_trees)
+                diffs = np.zeros(ts.num_trees)
+                for j, t in enumerate(ts.trees()):
+                    if discrete:
+                        span = np.ceil(t.interval[1]) - np.ceil(t.interval[0])
+                    else:
+                        span = t.span
+                    mean = mut_rate * span * t.total_branch_length
+                    N = t.num_mutations
+                    lower_pvals[j] = stats.poisson.cdf(mean, N)
+                    upper_pvals[j] = 1.0 - stats.poisson.cdf(mean, N + 1)
+                    # if we draw an indepenent Poisson with the same mean
+                    # it should be greater than observed half the time it is different
+                    # NOTE: more powerful might be a Wilcoxon test?
+                    rand = stats.poisson.rvs(mean, 1)
+                    diffs[j] = rand - N
+            self.sign_test(diffs)
+            self.bonferroni(lower_pvals)
+            self.bonferroni(upper_pvals)
+
+    def example_models(self):
+        print("BinaryMutations")
```

Comment:
I deleted these prints as I really dislike spam in the output of tests. I think a different structure is better anyway, allowing you to run the tests independently:

```
def test_binary_mutations(self):
    self.verify_model(msprime.BinaryMutations())

def test_jukes_cantor(self):
    self.verify_model(msprime.JukesCantory())

# etc
```

Much better to have lots of little tests.

Reply:
oh sounds good. sorry, I might have just squashed this change with a force-push; I'll fix it up.